features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/BaseCheckableTextViewMapperTest.kt

```diff
@@ -0,0 +1,287 @@
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.ArgumentMatchers.anyBoolean
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyVararg
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(ApiLevelExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal abstract class BaseCheckableTextViewMapperTest<T> :
+    LegacyBaseWireframeMapperTest() where T : TextView, T : Checkable {
+
+    private lateinit var testedCheckableTextViewMapper: CheckableTextViewMapper<T>
+
+    @Mock
+    lateinit var mockTextWireframeMapper: TextViewMapper<T>
+
+    @Forgery
+    lateinit var fakeTextWireframes: List<MobileSegment.Wireframe.TextWireframe>
+
+    @LongForgery
+    var fakeGeneratedIdentifier: Long = 0L
+
+    @Forgery
+    lateinit var fakeViewGlobalBounds: GlobalBounds
+
+    @Mock
+    lateinit var mockCheckableTextView: T
+
+    @Mock
+    lateinit var mockButtonDrawable: Drawable
+
+    @Mock
+    lateinit var mockConstantState: DrawableContainer.DrawableContainerState
+
+    @Mock
+    lateinit var mockCheckedConstantState: DrawableContainer.DrawableContainerState
+
+    @IntForgery(min = 0, max = 0xffffff)
+    var fakeCurrentTextColor: Int = 0
+
+    @StringForgery(regex = "#[0-9A-F]{8}")
+    lateinit var fakeCurrentTextColorString: String
+
+    @FloatForgery(min = 1f, max = 100f)
+    var fakeTextSize: Float = 1f
+
+    @IntForgery(min = CheckableCompoundButtonMapper.DEFAULT_CHECKABLE_HEIGHT_IN_DP, max = 100)
+    var fakeIntrinsicDrawableHeight = 1
+
+    @Mock
+    lateinit var mockCheckedDrawable: Drawable
+
+    @Mock
+    lateinit var mockNotCheckedDrawable: Drawable
+
+    @Mock
+    lateinit var mockClonedDrawable: Drawable
+
+    @IntForgery
+    var mockCloneDrawableIntrinsicHeight: Int = 0
+
+    @IntForgery
+    var mockCloneDrawableIntrinsicWidth: Int = 0
+
+    @BeforeEach
+    fun `set up`() {
+        mockClonedDrawable = mock {
+            whenever(it.intrinsicHeight) doReturn mockCloneDrawableIntrinsicHeight
+            whenever(it.intrinsicWidth) doReturn mockCloneDrawableIntrinsicWidth
+        }
+        mockCheckedConstantState = mock {
+            whenever(it.newDrawable(anyVararg(Resources::class))) doReturn mockClonedDrawable
```

Comment:
why `anyVarag` is used here instead of just `any<Resources>()`? I don't see any `newDrawable` method with `vararg` arguments.

Reply:
fixed, now I use a `mockResources` on this point.